contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+        require(!player.withdrawn, "Player has already withdrawn");
+        player.withdrawn = true;
+
+        // First player to withdraw redeems everyone's funds
+        if (!redeemed) {
+            redeemFromExternalPool(_minAmount);
+        }
+
+        uint256 payout = player.amountPaid;
+        uint256 playerIncentive = 0;
+        uint256 playerReward = 0;
+        uint256 playerCurveReward = 0;
+        if (player.mostRecentSegmentPaid == lastSegment.sub(1)) {
+            // Player is a winner and gets a bonus!
+            payout = payout.add(totalGameInterest.div(winnerCount));
+            playerReward = rewardsPerPlayer;
+            playerCurveReward = curveRewardsPerPlayer;
+            // If there's additional incentives, distributes them to winners
+            if (totalIncentiveAmount > 0) {
+                playerIncentive = totalIncentiveAmount.div(winnerCount);
+            }
+        }
+        emit Withdrawal(
+            msg.sender,
+            payout,
+            playerReward,
+            playerCurveReward,
+            playerIncentive
+        );
+
+        require(
+            IERC20(daiToken).transfer(msg.sender, payout),
+            "Fail to transfer ERC20 tokens on withdraw"
+        );
+
+        if (playerIncentive > 0) {
+            require(
+                IERC20(incentiveToken).transfer(msg.sender, playerIncentive),
+                "Fail to transfer ERC20 incentive tokens on withdraw"
+            );
+        }
+
+        if (playerReward > 0) {
+            require(
+                IERC20(matic).transfer(msg.sender, playerReward),
+                "Fail to transfer ERC20 rewards on withdraw"
+            );
+        }
+
+        if (playerCurveReward > 0) {
+            require(
+                IERC20(curve).transfer(msg.sender, playerCurveReward),
+                "Fail to transfer ERC20 rewards on withdraw"
+            );
+        }
+    }
+
+    /// @notice Redeems funds from the external pool and updates the internal accounting controls related to the game stats.
+    /// @dev Can only be called after the game is completed.
+    function redeemFromExternalPool(uint256 _minAmount)
+        public
+        whenGameIsCompleted
+    {
+        require(!redeemed, "Redeem operation already happened for the game");
+        redeemed = true;
+        uint256 lpBalance = gauge.balanceOf(address(this));
+        gauge.withdraw(lpBalance, true);
+
+        if (poolType == 0) {
+            pool.remove_liquidity_one_coin(
+                lpToken.balanceOf(address(this)),
+                inboundTokenIndexInt,
+                _minAmount,
+                true
+            );
+        } else if (poolType == 1) {
+            require(
+                lpToken.approve(
+                    address(pool),
+                    lpToken.balanceOf(address(this))
+                ),
+                "Fail to approve allowance to pool"
+            );
```

Comment:
interest to learn more about this - why do we need to approve an allowance to remove liquidity from pool type 1, but not needed for pool type 0?

Reply:
answered above